src/middlewared/middlewared/plugins/network.py

```diff
@@ -1019,17 +1023,26 @@ async def do_create(self, data):
             lag_id = None
             lagports_ids = []
             try:
-                async for i in self.__create_interface_datastore(data, {
-                    'interface': name,
-                }):
-                    interface_id = i
-
-                lag_id = await self.middleware.call(
-                    'datastore.insert',
-                    'network.lagginterface',
-                    {'lagg_interface': interface_id, 'lagg_protocol': data['lag_protocol'].lower()},
-                )
-                lagports_ids += await self.__set_lag_ports(lag_id, data['lag_ports'])
+                async for interface_id in self.__create_interface_datastore(data, {'interface': name}):
+                    lag_proto = data['lag_protocol'].lower()
+                    xmit = lacpdu_rate = None
+                    if lag_proto in ('lacp', 'loadbalance'):
+                        xmit = data['xmit_hash_policy'].lower() if data['xmit_hash_policy'] else None
+                        if xmit is None:
```

Comment:
Is this required, we can just perhaps add the default value above in the inline if ?